astropy/cosmology/funcs.py

```diff
@@ -34,20 +36,52 @@ def z_at_value(func, fval, zmin=1e-8, zmax=1000, ztol=1e-8, maxfun=500):
     ----------
     func : function or method
        A function that takes a redshift as input.
+
     fval : `~astropy.units.Quantity` instance
-       The value of ``func(z)``.
+       The (scalar) value of ``func(z)`` to recover.
+
     zmin : float, optional
        The lower search limit for ``z``.  Beware of divergences
        in some cosmological functions, such as distance moduli,
        at z=0 (default 1e-8).
+
     zmax : float, optional
        The upper search limit for ``z`` (default 1000).
+
     ztol : float, optional
        The relative error in ``z`` acceptable for convergence.
+
     maxfun : int, optional
        The maximum number of function evaluations allowed in the
        optimization routine (default 500).
 
+    method : str or callable, optional
+       Type of solver to pass to the minimizer. The built-in options provided by
+       :fun:`scipy.optimize.minimize_scalar` are 'Brent' (default), 'Golden' and
+       'Bounded' with names case insensitive - see documentation there for details.
+       It also accepts a custom solver by passing any user-provided callable object
+       that meets the requirements listed under the Notes on "Custom minimizers"
+       therein, and in more detail in :doc:`scipy:reference/tutorial/optimize`;
+       although their use is currently untested.
+
+       .. versionadded:: 4.3
+
+    bracket : sequence, optional
+       For methods 'Brent' and 'Golden', ``bracket`` defines the bracketing
+       interval and can either have three items (z1, z2, z3) so that z1 < z2 < z3
+       and ``func(z2) < func(z1), func(z3)`` or two items z1 and z3 which are
+       assumed to be a starting interval for a downhill bracket search.
+       For bimodal functions such as angular diameter distance this can be
```

Comment:
Do you mean non-monotonic functions?

Reply:
Yes, probably had my head in statistical distributions at the time (I have to reconstruct this stuff myself from last year...)

Reply:
I know the feeling. I have some PRs languishing that will need a careful read-through to get myself back up to speed. 🔍